src/components/Header/Header.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import { Error } from '../../types/Error';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  newTodoField: React.RefObject<HTMLInputElement>;
+  addTodoToServer: (arg0: string) => Promise<void>;
+  isAdding: boolean;
+  setErrorMessage: (arg0: Error) => void;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  newTodoField,
+  addTodoToServer,
+  isAdding,
+  setErrorMessage,
+}) => {
+  const [title, setTitle] = useState('');
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setErrorMessage({
+        hasMessage: 'Title can\'t be empty',
+      });
+
+      return;
+    }
+
+    addTodoToServer(title);
+    setTitle('');
```

Comment:
Better await for addTodoToServer is completed successfully. Now you clear input before result.